File: python/atbash-cipher/atbash_cipher.py
import string


class Atbash:

    PLAIN = 'abcdefghijklmnopqrstuvwxyz'
    PRIME = 'zyxwvutsrqponmlkjihgfedcba'
    CIPHER = dict(zip(list(PLAIN), list(PRIME)))
    EXCLUDE = set(string.punctuation + ' ')

    @staticmethod
    def encode(self, msg):
        return self.split_every_five(self, self.encoded(self, msg))

    @staticmethod
    def split_every_five(self, encoded):
        return ' '.join([encoded[i:i + 5] for i in range(0, len(encoded), 5)])

    @staticmethod
    def encoded(self, msg):
        return ''.join(([char if char.isdigit() else self.CIPHER[char]
                         for char in self.clean(self, msg)]))

    @staticmethod
    def clean(self, msg):
        return (char for char in msg.lower() if char not in self.EXCLUDE)

    @staticmethod
    def decode(self, msg):
        return self.encoded(self, msg)


def encode(msg):
    return Atbash.encode(Atbash, msg)


def decode(msg):
    return Atbash.decode(Atbash, msg)

File: go/atbash-cipher/atbash_cipher.go
package atbash

import (
	"math"
	"strings"
	"unicode"
)

// Plain:  abcdefghijklmnopqrstuvwxyz
// Cipher: zyxwvutsrqponmlkjihgfedcba
var atbash = map[rune]rune{
	'a': 'z',
	'b': 'y',
	'c': 'x',
	'd': 'w',
	'e': 'v',
	'f': 'u',
	'g': 't',
	'h': 's',
	'i': 'r',
	'j': 'q',
	'k': 'p',
	'l': 'o',
	'm': 'n',
	'n': 'm',
	'o': 'l',
	'p': 'k',
	'q': 'j',
	'r': 'i',
	's': 'h',
	't': 'g',
	'u': 'f',
	'v': 'e',
	'w': 'd',
	'x': 'c',
	'y': 'b',
	'z': 'a',
}

func Atbash(message string) (ciphertext string) {
	for _, r := range removeNonAlphanumeric(strings.ToLower(message)) {
		ciphertext += string(encode(r))
	}
	return strings.Join(splitEveryN(ciphertext, 5), " ")
}

func removeNonAlphanumeric(s string) (result string) {
	for _, r := range s {
		if unicode.IsDigit(r) || unicode.IsLetter(r) {
			result += string(r)
		}
	}
	return result
}

func splitEveryN(message string, n int) (groups []string) {
	for i := 0; i < len(message); i += n {
		upperBound := math.Min(float64(len(message)), float64(i+n))
		group := message[i:int(upperBound)]
		groups = append(groups, group)
	}
	return groups
}

func encode(r rune) rune {
	if unicode.IsLetter(r) {
		return atbash[r]
	}
	return r
}
